--- xpdf-3.00/splash/SplashFTFont.cc.orig	2004-01-22 02:26:44.000000000 +0100
+++ xpdf-3.00/splash/SplashFTFont.cc	2004-01-26 12:57:05.000000000 +0100
@@ -154,22 +154,22 @@
   }
 
   // if we have the FT2 bytecode interpreter, autohinting won't be used
-#ifdef TT_CONFIG_OPTION_BYTECODE_INTERPRETER
+//#ifdef TT_CONFIG_OPTION_BYTECODE_INTERPRETER
   if (FT_Load_Glyph(ff->face, gid,
 		    aa ? FT_LOAD_NO_BITMAP : FT_LOAD_DEFAULT)) {
     return gFalse;
   }
-#else
+//#else
   // FT2's autohinting doesn't always work very well (especially with
   // font subsets), so turn it off if anti-aliasing is enabled; if
   // anti-aliasing is disabled, this seems to be a tossup - some fonts
   // look better with hinting, some without, so leave hinting on
-  if (FT_Load_Glyph(ff->face, gid,
-		    aa ? FT_LOAD_NO_HINTING | FT_LOAD_NO_BITMAP
-                       : FT_LOAD_DEFAULT)) {
-    return gFalse;
-  }
-#endif
+//  if (FT_Load_Glyph(ff->face, gid,
+//		    aa ? FT_LOAD_NO_HINTING | FT_LOAD_NO_BITMAP
+//                      : FT_LOAD_DEFAULT)) {
+//    return gFalse;
+//  }
+//#endif
   if (FT_Render_Glyph(slot, aa ? ft_render_mode_normal
 		               : ft_render_mode_mono)) {
     return gFalse;
