--- xpdf-3.00/splash/SplashT1Font.h.gcc401	2004-01-22 02:26:44.000000000 +0100
+++ xpdf-3.00/splash/SplashT1Font.h	2005-08-03 12:11:51.000000000 +0200
@@ -21,6 +21,8 @@
 // SplashT1Font
 //------------------------------------------------------------------------
 
+class SplashT1FontFile;
+
 class SplashT1Font: public SplashFont {
 public:
 
--- xpdf-3.00/xpdf/TextOutputDev.h.gcc401	2004-01-22 02:26:45.000000000 +0100
+++ xpdf-3.00/xpdf/TextOutputDev.h	2005-08-03 12:16:22.000000000 +0200
@@ -166,6 +166,8 @@
 // TextLine
 //------------------------------------------------------------------------
 
+class TextBlock;
+
 class TextLine {
 public:
 
@@ -223,6 +225,8 @@
 // TextBlock
 //------------------------------------------------------------------------
 
+class TextPage;
+
 class TextBlock {
 public:
 
@@ -338,6 +342,8 @@
 
 #endif // TEXTOUT_WORD_LIST
 
+class TextLineFrag;
+
 //------------------------------------------------------------------------
 // TextPage
 //------------------------------------------------------------------------
